File: app/src/main/java/com/routee/qianbaotest/base/BaseActivity.java
package com.routee.qianbaotest.base;

import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by hdb on 2017/8/7.
 * description: ${TODO}
 */

public abstract class BaseActivity extends AppCompatActivity {

    private Unbinder mButterKnife;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getRootView());
        mButterKnife = ButterKnife.bind(this);
        initView();
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
    }

    public abstract int getRootView();

    public abstract void initView();

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mButterKnife.unbind();
    }
}

File: app/src/main/java/com/routee/qianbaotest/base/BaseActivity.kt
package com.routee.qianbaotest.base

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

/**
 * Created by hdb on 2017/8/7.
 * description: ${TODO}
 */

abstract class BaseActivity : AppCompatActivity() {


    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(rootView())
        initView()
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
    }

    abstract fun rootView(): Int

    abstract fun initView()

    override fun onDestroy() {
        super.onDestroy()
    }
}
